packages/waas/src/payloads/signers/secp256r1.ts

```diff
@@ -0,0 +1,61 @@
+import { ethers } from "ethers";
+import { PayloadSigner } from "./index";
+
+export async function newSECP256R1Signer(privateKey: string): Promise<PayloadSigner> {
+  const [pubKey, privKey] = await loadSECP256R1Keys(privateKey)
+
+  const exPrivateKey = await window.crypto.subtle.exportKey('jwk', privKey)
+  const exPublicKey = await window.crypto.subtle.exportKey('jwk', pubKey)
+  const keys = [exPublicKey, exPrivateKey]
+
+  const encoder = new TextEncoder()
+  return {
+    privateKey: ethers.utils.hexlify(encoder.encode(JSON.stringify(keys))),
```

Comment:
btw, the advantage of the r1 curve is we can great a key in window.crypto that is not exportable, and never gain access to the private key. This is what makes it more secure. Therefore, the privateKey wouldn't be available here. But you should return the hex of the public key.

Lmk what you thoughts are though...?

Reply:
for the r1 publicKey(): Promise<string> method, we can use `ethers.utils.hexlify( window.crypto.subtle.exportKey('raw', pubKey))` which will return 130 character length string in hex of the 65 byte public key.

potentially to.... maybe we should add `address: Promise<string>` to the SessionSigner interface, for both k1 and r1 of course.. where k1 already has address, and for r1 address we could do like `'0x'+publicKey().slice(0,40)`

and we have the `alg()` to tell us which curve it is..

in terms of "verify()" if we have just the alg, address, message and signature... I

Reply:
It uses export raw for ``publicKey``.  I don't like the name ``sessionAddress`` as the address makes sense only in ethereum. So I would rather use names such as ``sessionId``, ``sessionKeyType``, ``sessionPublicKey``, ``signature``.

Reply:
agreed, lets go with the name `sessionId`